[Bug]: ECF files parsing removes spaces in strings

### FAQ check

- [X] Yes, I checked the FAQ and my question has not been addressed.

### Instrument

Other (any stage)

### What happened?

When specifying a filepath in an ECS file that contains spaces, for example `/home/user/my data` the program fails giving back the error:
```
WARNING: Unable to find an output metadata file from Eureka!'s S1 in the folder:
"/home/user/mydata/"
Assuming this S1 data was produced by the JWST pipeline instead.
Traceback (most recent call last):
  File "/home/user/eureka/run_eureka.py", line 22, in <module>
    s2_meta = s2.calibrateJWST(eventlabel, ecf_path=ecf_path)
  File "/home/user/miniconda3/envs/eureka/lib/python3.9/site-packages/eureka/S2_calibrations/s2_calibrate.py", line 82, in calibrateJWST
    meta = util.find_fits(meta)
  File "home/user/miniconda3/envs/eureka/lib/python3.9/site-packages/eureka/lib/util.py", line 339, in find_fits
    raise AssertionError(message)
AssertionError: Unable to find any "rateints.fits" files in the inputdir: 
"/home/user/mydata/"!
You likely need to change the inputdir in S2_nirspec_fs_template.ecf to point to the folder containing the "rateints.fits" files.
```

The parsing function strips all whitespace characters from the string. Using quotes or backspaces does not work. Looking at the parsing function `MetaClass.Read` in `eureka/lib/util.py` it seems like [these lines](https://github.com/kevin218/Eureka/blob/3389d9b56dd632f2568c8372b1dc340318e7393d/src/eureka/lib/readECF.py#L181-L183) are the culprit:

```
        for line in cleanlines:
            name = line.split()[0]
            val = ''.join(line.split()[1:])
```

I agree with the suggestion in #239 as using an existing parser for an established config file format would probably be the most robust approach. One could of course also change the parser such that spaces after the initial split between parameter name and value are conserved. Happy to help with a small bug fix.

### What operating system are you using?

Linux Ubuntu 22.10

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/docs/source/ecf.rst
+++ b/docs/source/ecf.rst
@@ -842,7 +842,7 @@ where starry replaces both the batman_tr and batman_ecl models and offers a more
 
 manual_clip
 '''''''''''
-Optional. A list of lists specifying the start and end integration numbers for manual removal. E.g., to remove the first 20 data points specify [[0,20]], and to also remove the last 20 data points specify [[0,20],[-20,None]].
+Optional. A list of lists specifying the start and end integration numbers for manual removal. E.g., to remove the first 20 data points specify [[0,20]], and to also remove the last 20 data points specify [[0,20],[-20,None]]. If you want to clip the 10th integration, this would be index 9 since python uses zero-indexing. And the manual_clip start and end values are used to slice a numpy array, so they follow the same convention of *inclusive* start index and *exclusive* end index. In other words, to trim the 10th integrations, you would set manual_clip to [[9,10]].
 
 
 Limb Darkening Parameters

--- a/docs/source/ecf.rst
+++ b/docs/source/ecf.rst
@@ -172,11 +172,11 @@ Number of rows to use for ROEBA routine along the bottom of the subarray
 
 topdir + inputdir
 '''''''''''''''''
-The path to the directory containing the Stage 0 JWST data (uncal.fits).
+The path to the directory containing the Stage 0 JWST data (uncal.fits). Directories containing spaces should be enclosed in quotation marks.
 
 topdir + outputdir
 ''''''''''''''''''
-The path to the directory in which to output the Stage 1 JWST data and plots.
+The path to the directory in which to output the Stage 1 JWST data and plots. Directories containing spaces should be enclosed in quotation marks.
 
 testing_S1
 ''''''''''
@@ -253,12 +253,12 @@ If True, plots will automatically be closed rather than popping up on the screen
 
 topdir + inputdir
 '''''''''''''''''
-The path to the directory containing the Stage 1 JWST data.
+The path to the directory containing the Stage 1 JWST data. Directories containing spaces should be enclosed in quotation marks.
 
 
 topdir + outputdir
 ''''''''''''''''''
-The path to the directory in which to output the Stage 2 JWST data and plots.
+The path to the directory in which to output the Stage 2 JWST data and plots. Directories containing spaces should be enclosed in quotation marks.
 
 
 
@@ -594,23 +594,23 @@ If True, more details will be printed about steps.
 
 topdir + inputdir
 '''''''''''''''''
-The path to the directory containing the Stage 2 JWST data. For HST observations, the sci_dir and cal_dir folders will only be checked if this folder does not contain FITS files.
+The path to the directory containing the Stage 2 JWST data. For HST observations, the sci_dir and cal_dir folders will only be checked if this folder does not contain FITS files. Directories containing spaces should be enclosed in quotation marks.
 
 topdir + inputdir + sci_dir
 '''''''''''''''''''''''''''
-Optional, only used for HST analyses. The path to the folder containing the science spectra. Defaults to 'sci'.
+Optional, only used for HST analyses. The path to the folder containing the science spectra. Defaults to 'sci'. Directories containing spaces should be enclosed in quotation marks.
 
 topdir + inputdir + cal_dir
 '''''''''''''''''''''''''''
-Optional, only used for HST analyses. The path to the folder containing the wavelength calibration imaging mode observations. Defaults to 'cal'.
+Optional, only used for HST analyses. The path to the folder containing the wavelength calibration imaging mode observations. Defaults to 'cal'. Directories containing spaces should be enclosed in quotation marks.
 
 topdir + outputdir
 ''''''''''''''''''
-The path to the directory in which to output the Stage 3 JWST data and plots.
+The path to the directory in which to output the Stage 3 JWST data and plots. Directories containing spaces should be enclosed in quotation marks.
 
 topdir + time_file
 ''''''''''''''''''
-Optional. The path to a file that contains the time array you want to use instead of the one contained in the FITS file.
+Optional. The path to a file that contains the time array you want to use instead of the one contained in the FITS file. Directories containing spaces should be enclosed in quotation marks.
 
 
 
@@ -790,12 +790,12 @@ If True, more details will be printed about steps.
 
 topdir + inputdir
 '''''''''''''''''
-The path to the directory containing the Stage 3 JWST data.
+The path to the directory containing the Stage 3 JWST data. Directories containing spaces should be enclosed in quotation marks.
 
 
 topdir + outputdir
 ''''''''''''''''''
-The path to the directory in which to output the Stage 4 JWST data and plots.
+The path to the directory in which to output the Stage 4 JWST data and plots. Directories containing spaces should be enclosed in quotation marks.
 
 
 
@@ -965,12 +965,12 @@ If True, plots will automatically be closed rather than popping up on the screen
 
 topdir + inputdir
 '''''''''''''''''
-The path to the directory containing the Stage 4 JWST data.
+The path to the directory containing the Stage 4 JWST data. Directories containing spaces should be enclosed in quotation marks.
 
 
 topdir + outputdir
 ''''''''''''''''''
-The path to the directory in which to output the Stage 5 JWST data and plots.
+The path to the directory in which to output the Stage 5 JWST data and plots. Directories containing spaces should be enclosed in quotation marks.
 
 
 Stage 5 Fit Parameters
@@ -1173,11 +1173,11 @@ If True, plots will automatically be closed rather than popping up on the screen
 
 topdir + inputdir
 '''''''''''''''''
-The path to the directory containing the Stage 5 JWST data.
+The path to the directory containing the Stage 5 JWST data. Directories containing spaces should be enclosed in quotation marks.
 
 topdir + outputdir
 ''''''''''''''''''
-The path to the directory in which to output the Stage 6 JWST data and plots.
+The path to the directory in which to output the Stage 6 JWST data and plots. Directories containing spaces should be enclosed in quotation marks.
 
 topdir + model_spectrum
 '''''''''''''''''''''''
@@ -1185,7 +1185,7 @@ The path to a model spectrum to plot underneath the observations to show how the
 compare to the input model for simulated observations or how the fitted results compare to a
 retrieved model for real observations. Set to None if no model should be plotted.
 The file should have column 1 as the wavelength and column 2 should contain the transmission
-or emission spectrum. Any headers must be preceded by a #.
+or emission spectrum. Any headers must be preceded by a #. Directories containing spaces should be enclosed in quotation marks.
 
 model_x_unit
 ''''''''''''

--- a/setup.cfg
+++ b/setup.cfg
@@ -42,7 +42,7 @@ install_requires =
     exotic-ld
     george # Needed for GP
     gwcs
-    h5py < 3.2
+    h5py
     jsonschema < 4.10.0 # 4.10.0 breaks S2 ASDF stuff for some reason
     lmfit
     matplotlib

--- a/src/eureka/S1_detector_processing/bias_sub.py
+++ b/src/eureka/S1_detector_processing/bias_sub.py
@@ -30,6 +30,8 @@ def do_correction(input_model, bias_model, meta, log):
     output_model: data model object
         bias-subtracted science data
     """
+    log.writelog('Doing super-bias subtraction.')
+
     # Check for subarray mode and extract subarray from the
     # bias reference data if necessary
     if not reffile_utils.ref_matches_sci(input_model, bias_model):

--- a/src/eureka/S1_detector_processing/group_level.py
+++ b/src/eureka/S1_detector_processing/group_level.py
@@ -32,7 +32,7 @@ def GLBS(input_model, log, meta):
         The input group-level data product with background removed.
     '''
 
-    log.writelog('  Running Group Level Background Subtraction.')
+    log.writelog('Running Group Level Background Subtraction.')
 
     all_data = input_model.data
     dq = input_model.groupdq
@@ -51,6 +51,8 @@ def GLBS(input_model, log, meta):
         meta.isrotate = False
 
     for ngrp in range(all_data.shape[1]):
+        log.writelog(f'  Starting group {ngrp}.')
+
         grp_data = all_data[:, ngrp, :, :]
         grp_mask = np.ones(grp_data.shape, dtype=bool)
         dqmask = np.where((dq[:, ngrp, :, :] % 2 == 1) |
@@ -103,7 +105,7 @@ def mask_trace(input_model, log, meta):
     input_model : jwst.datamodels.QuadModel
         The input group-level data product with trace mask object
     '''
-    log.writelog('  Masking Curved Trace.')
+    log.writelog('  Masking curved trace.')
 
     all_data = input_model.data
 

--- a/src/eureka/S1_detector_processing/s1_process.py
+++ b/src/eureka/S1_detector_processing/s1_process.py
@@ -163,6 +163,11 @@ class EurekaS1Pipeline(Detector1Pipeline):
         - February 2022 Aarynn Carter /  Eva-Maria Ahrer
             Updated for JWST version 1.3.3, code restructure
         '''
+        # Define superbias offset procedure
+        self.superbias = Eureka_SuperBiasStep()
+        self.superbias.s1_meta = meta
+        self.superbias.s1_log = log
+
         # Figure out which instrument we're working on
         with fits.open(filename) as f:
             instrument = f[0].header['INSTRUME']
@@ -201,11 +206,6 @@ class EurekaS1Pipeline(Detector1Pipeline):
             self.firstframe.skip = meta.skip_firstframe
             self.lastframe.skip = meta.skip_lastframe
             self.rscd.skip = meta.skip_rscd
-
-        # Define superbias offset procedure
-        self.superbias = Eureka_SuperBiasStep()
-        self.superbias.s1_meta = meta
-        self.superbias.s1_log = log
 
         # Define ramp fitting procedure
         self.ramp_fit = Eureka_RampFitStep()

--- a/src/eureka/S1_detector_processing/s1_process.py
+++ b/src/eureka/S1_detector_processing/s1_process.py
@@ -162,7 +162,12 @@ class EurekaS1Pipeline(Detector1Pipeline):
         - February 2022 Aarynn Carter /  Eva-Maria Ahrer
             Updated for JWST version 1.3.3, code restructure
         '''
-        # Run the pipeline
+        # Over-write the superbias offset function
+        self.superbias = Eureka_SuperBiasStep()
+        self.superbias.s1_meta = meta
+        self.superbias.s1_log = log
+
+        # Figure out which instrument we're working on
         with fits.open(filename) as f:
             instrument = f[0].header['INSTRUME']
 
@@ -201,11 +206,6 @@ class EurekaS1Pipeline(Detector1Pipeline):
             self.lastframe.skip = meta.skip_lastframe
             self.rscd.skip = meta.skip_rscd
 
-        # Define superbias offset procedure
-        self.superbias = Eureka_SuperBiasStep()
-        self.superbias.s1_meta = meta
-        self.superbias.s1_log = log
-        
         # Define ramp fitting procedure
         self.ramp_fit = Eureka_RampFitStep()
         self.ramp_fit.algorithm = meta.ramp_fit_algorithm

--- a/src/eureka/S1_detector_processing/update_saturation.py
+++ b/src/eureka/S1_detector_processing/update_saturation.py
@@ -25,20 +25,20 @@ def update_sat(input_model, log, meta):
     input_model : jwst.datamodels.QuadModel
         The input group-level data product with updated saturation flags.
     '''
-    log.writelog('  Applying a more severe saturation flagging.')
+    log.writelog('Applying a more severe saturation flagging.')
 
     # Compute the median of the groupdq map
     if meta.dq_sat_mode == "percentile":
-        log.writelog('Creating a saturation map based on the\n' +
+        log.writelog('  Creating a saturation map based on the ' +
                      'percentile set in the ecf file.')
         median_sat = np.percentile(input_model.groupdq, 
                                    meta.dq_sat_percentile, axis=0).astype(int)
     elif meta.dq_sat_mode == "min":
-        log.writelog('Creating a saturation map based on the\n' +
+        log.writelog('  Creating a saturation map based on the ' +
                      'minimum saturated group for each pixel.')
         median_sat = np.max(input_model.groupdq, axis=0).astype(int)    
     elif meta.dq_sat_mode == "defined":
-        log.writelog('Creating a saturation map based on the\n' +
+        log.writelog('  Creating a saturation map based on the ' +
                      'user defined columns.')
         median_sat = np.zeros_like(np.median(input_model.groupdq, axis=0))
         
@@ -47,7 +47,7 @@ def update_sat(input_model, log, meta):
     median_sat_mask = (median_sat == sat_flag)
  
     # Expand saturated pixels to full columns
-    log.writelog('    Expand flags along columns.')
+    log.writelog('  Expand flags along columns.')
     new_sat_mask = 1 * median_sat_mask
     ngrp = new_sat_mask.shape[0]
     ncols = new_sat_mask.shape[1]
@@ -64,21 +64,21 @@ def update_sat(input_model, log, meta):
             
     # Expand saturation flags to one group before
     if meta.expand_prev_group:
-        log.writelog('    Expand flags to previous group.')
+        log.writelog('  Expand flags to previous group.')
         for i in range(ngrp-1):
             new_sat_mask[i, :, :] += new_sat_mask[i+1, :, :] 
 
     # Make sure that we did not put saturation in Group 1
     # If so, remove it (o.w. no info at all for ramp_fitting)
     if np.count_nonzero(new_sat_mask[0]) > 0:
-        log.writelog('    WARNING')
+        log.writelog('  WARNING:')
         log.writelog('    Some saturated flags found in the first group.')
         log.writelog('    Removing the flags so the first group is fully used')
         new_sat_mask[0, :, :] *= False
 
     # If flags in the second group, just raise a warning
     if np.count_nonzero(new_sat_mask[1]) > 0:
-        log.writelog('    WARNING')
+        log.writelog('  WARNING:')
         log.writelog('    Saturation flags found in the 2nd group ')
         log.writelog('    This means some columns only have nGroup=1... ')
         log.writelog('    Use caution for this part of the detector.')

--- a/src/eureka/S3_data_reduction/plots_s3.py
+++ b/src/eureka/S3_data_reduction/plots_s3.py
@@ -426,7 +426,7 @@ def subdata(meta, i, n, m, subdata, submask, expected, loc, variance):
         plt.pause(0.1)
 
 
-def driftypos(data, meta):
+def driftypos(data, meta, m):
     '''Plot the spatial jitter. (Fig 3104)
 
     Parameters
@@ -435,6 +435,8 @@ def driftypos(data, meta):
         The Dataset object.
     meta : eureka.lib.readECF.MetaClass
         The metadata object.
+    m : int
+        The file number.
 
     Notes
     -----
@@ -449,13 +451,15 @@ def driftypos(data, meta):
     plt.ylabel('Spectrum spatial profile center')
     plt.xlabel('Integration Number')
     plt.tight_layout()
-    fname = 'figs'+os.sep+'fig3104_DriftYPos'+plots.figure_filetype
+    file_number = str(m).zfill(int(np.floor(np.log10(meta.num_data_files))+1))
+    fname = (f'figs{os.sep}fig3104_file{file_number}_DriftYPos' +
+             plots.figure_filetype)
     plt.savefig(meta.outputdir+fname, bbox_inches='tight', dpi=300)
     if not meta.hide_plots:
         plt.pause(0.2)
 
 
-def driftywidth(data, meta):
+def driftywidth(data, meta, m):
     '''Plot the spatial profile's fitted Gaussian width. (Fig 3105)
 
     Parameters
@@ -464,6 +468,8 @@ def driftywidth(data, meta):
         The Dataset object.
     meta : eureka.lib.readECF.MetaClass
         The metadata object.
+    m : int
+        The file number.
 
     Notes
     -----
@@ -478,7 +484,9 @@ def driftywidth(data, meta):
     plt.ylabel('Spectrum spatial profile width')
     plt.xlabel('Integration Number')
     plt.tight_layout()
-    fname = 'figs'+os.sep+'fig3105_DriftYWidth'+plots.figure_filetype
+    file_number = str(m).zfill(int(np.floor(np.log10(meta.num_data_files))+1))
+    fname = (f'figs{os.sep}fig3105_file{file_number}_DriftYWidth' +
+             plots.figure_filetype)
     plt.savefig(meta.outputdir+fname, bbox_inches='tight', dpi=300)
     if not meta.hide_plots:
         plt.pause(0.2)

--- a/src/eureka/S3_data_reduction/s3_reduce.py
+++ b/src/eureka/S3_data_reduction/s3_reduce.py
@@ -387,8 +387,8 @@ def reduce(eventlabel, ecf_path=None, s2_meta=None, input_meta=None):
                                                           m, integ=None)
                         if meta.isplots_S3 >= 1:
                             # make y position and width plots
-                            plots_s3.driftypos(data, meta)
-                            plots_s3.driftywidth(data, meta)
+                            plots_s3.driftypos(data, meta, m)
+                            plots_s3.driftywidth(data, meta, m)
 
                     # Select only aperture region
                     apdata, aperr, apmask, apbg, apv0 = inst.cut_aperture(data,

--- a/src/eureka/S4_generate_lightcurves/generate_LD.py
+++ b/src/eureka/S4_generate_lightcurves/generate_LD.py
@@ -47,7 +47,7 @@ def exotic_ld(meta, spec, log, white=False):
     custom_wavelengths = None
     custom_throughput = None
 
-    if hasattr(meta, 'exotic_ld_file'):
+    if hasattr(meta, 'exotic_ld_file') and meta.exotic_ld_file is not None:
         mode = 'custom'
         log.writelog("Using custom throughput file " +
                      meta.exotic_ld_file,

--- a/src/eureka/S4_generate_lightcurves/generate_LD.py
+++ b/src/eureka/S4_generate_lightcurves/generate_LD.py
@@ -1,8 +1,6 @@
 from exotic_ld import StellarLimbDarkening
 import numpy as np
 import pandas as pd
-import glob
-import os
 
 
 def exotic_ld(meta, spec, log, white=False):
@@ -35,13 +33,6 @@ def exotic_ld(meta, spec, log, white=False):
 
     log.writelog("...using exotic-ld package...",
                  mute=(not meta.verbose))
-
-    # Check if exotic-ld directory includes csv files
-    exotic_ld_files = glob.glob(meta.exotic_ld_direc + os.sep + '**' + os.sep +
-                                '*.flx', recursive=True)
-    if not exotic_ld_files:
-        raise AssertionError('Unable to find ancillary files.' +
-                             'Have you downloaded them? (see Zenodo link)')
 
     # Set the observing mode
     custom_wavelengths = None

--- a/src/eureka/S4_generate_lightcurves/s4_genLC.py
+++ b/src/eureka/S4_generate_lightcurves/s4_genLC.py
@@ -154,7 +154,14 @@ def genlc(eventlabel, ecf_path=None, s3_meta=None, input_meta=None):
             elif meta.wave_min < np.min(wave_1d):
                 log.writelog(f'WARNING: The selected meta.wave_min '
                              f'({meta.wave_min}) is smaller than the shortest '
-                             f'wavelength ({np.min(wave_1d)})')
+                             f'wavelength ({np.min(wave_1d)})!!')
+                if meta.inst == 'miri':
+                    axis = 'ywindow'
+                else:
+                    axis = 'xwindow'
+                log.writelog('  If you want to use wavelengths shorter than '
+                             f'{np.min(wave_1d)}, you will need to decrease '
+                             f'your {axis} lower limit in Stage 3.')
             if meta.wave_max is None:
                 meta.wave_max = np.max(wave_1d)
                 log.writelog(f'No value was provided for meta.wave_max, so '
@@ -163,7 +170,14 @@ def genlc(eventlabel, ecf_path=None, s3_meta=None, input_meta=None):
             elif meta.wave_max > np.max(wave_1d):
                 log.writelog(f'WARNING: The selected meta.wave_max '
                              f'({meta.wave_max}) is larger than the longest '
-                             f'wavelength ({np.max(wave_1d)})')
+                             f'wavelength ({np.max(wave_1d)})!!')
+                if meta.inst == 'miri':
+                    axis = 'ywindow'
+                else:
+                    axis = 'xwindow'
+                log.writelog('  If you want to use wavelengths longer than '
+                             f'{np.max(wave_1d)}, you will need to increase '
+                             f'your {axis} upper limit in Stage 3.')
 
             if meta.photometry:
                 meta.n_int, meta.subnx = spec.aplev.shape[0], 1

--- a/src/eureka/S4_generate_lightcurves/s4_genLC.py
+++ b/src/eureka/S4_generate_lightcurves/s4_genLC.py
@@ -164,9 +164,6 @@ def genlc(eventlabel, ecf_path=None, s3_meta=None, input_meta=None):
                 log.writelog(f'WARNING: The selected meta.wave_max '
                              f'({meta.wave_max}) is larger than the longest '
                              f'wavelength ({np.max(wave_1d)})')
-            indices = np.logical_and(wave_1d >= meta.wave_min,
-                                     wave_1d <= meta.wave_max)
-            wave_1d_trimmed = wave_1d[indices]
 
             if meta.photometry:
                 meta.n_int, meta.subnx = spec.aplev.shape[0], 1
@@ -381,7 +378,7 @@ def genlc(eventlabel, ecf_path=None, s3_meta=None, input_meta=None):
             log.writelog(f"Stage 4 MAD = {np.round(meta.mad_s4, 2):.2f} ppm")
             if not meta.photometry:
                 if meta.isplots_S4 >= 1:
-                    plots_s4.lc_driftcorr(meta, wave_1d_trimmed, spec.optspec,
+                    plots_s4.lc_driftcorr(meta, wave_1d, spec.optspec,
                                           optmask=spec.optmask)
 
             log.writelog("Generating light curves")

--- a/src/eureka/S5_lightcurve_fitting/s5_fit.py
+++ b/src/eureka/S5_lightcurve_fitting/s5_fit.py
@@ -254,6 +254,8 @@ def fitlc(eventlabel, ecf_path=None, s4_meta=None, input_meta=None):
                     except FileNotFoundError:
                         raise Exception("The limb-darkening file "
                                         f"{ld_fix_file} could not be found.")
+                    if len(ld_coeffs.shape) == 1:
+                        ld_coeffs = ld_coeffs[np.newaxis, :]
                 else:
                     ld_coeffs = None
 
@@ -304,6 +306,8 @@ def fitlc(eventlabel, ecf_path=None, s4_meta=None, input_meta=None):
                 except FileNotFoundError:
                     raise Exception("The limb-darkening file " + ld_fix_file +
                                     " could not be found.")
+                if len(ld_coeffs.shape) == 1:
+                    ld_coeffs = ld_coeffs[np.newaxis, :]
             else:
                 ld_coeffs = None
 

--- a/src/eureka/S6_planet_spectra/s6_spectra.py
+++ b/src/eureka/S6_planet_spectra/s6_spectra.py
@@ -330,7 +330,9 @@ def plot_spectra(eventlabel, ecf_path=None, s5_meta=None, input_meta=None):
                                         scale_height, meta.planet_R0)
 
                 save_table(meta, log)
-                convert_s5_LC(meta, log)
+            
+            # Copy S5 text files to a single h5 file
+            convert_s5_LC(meta, log)
 
             # make citations for current stage
             util.make_citations(meta, 6)

--- a/src/eureka/lib/readECF.py
+++ b/src/eureka/lib/readECF.py
@@ -1,5 +1,5 @@
 import os
-
+import shlex
 # Required in case user passes in a numpy object (e.g. np.inf)
 import numpy as np
 
@@ -179,8 +179,9 @@ class MetaClass:
                 cleanlines.append(line)
 
         for line in cleanlines:
-            name = line.split()[0]
-            val = ''.join(line.split()[1:])
+            name = shlex.split(line)[0]
+            # Split off the name and remove all spaces except quoted substrings
+            val = ''.join(shlex.split(line)[1:])
             try:
                 val = eval(val)
             except:

--- a/src/eureka/lib/readECF.py
+++ b/src/eureka/lib/readECF.py
@@ -181,7 +181,9 @@ class MetaClass:
         for line in cleanlines:
             name = shlex.split(line)[0]
             # Split off the name and remove all spaces except quoted substrings
-            val = ''.join(shlex.split(line)[1:])
+            # Also keep quotation marks for things that need to be escaped
+            # (e.g. max is a built-in funciton)
+            val = ''.join(shlex.split(line, posix=False)[1:])
             try:
                 val = eval(val)
             except:

